### ISSUE TITLE
JIT-traced module incurs bug: nSubTensors_ > 1 ASSERT FAILED 

### ISSUE THREAD

**Issue by msieb1** [pytorch/pytorch#37511]
## 🐛 Bug

After JIT-tracing a module via `torch.jit.trace`, the resulting model throws the following error trace when running it on a sample input:

## Error Trace

```
RuntimeError: nSubTensors_ > 1 ASSERT FAILED at /pytorch/torch/csrc/jit/fuser/partition_desc.h:24, please report a bug to PyTorch. (PartitionDesc at /pytorch/torch/csrc/jit/fuser/partition_desc.h:24)
frame #0: std::function<std::string ()>::operator()() const + 0x11 (0x7f2223c0f441 in /opt/conda/envs/lab42/lib/python3.7/site-packages/torch/lib/libc10.so)
frame #1: c10::Error::Error(c10::SourceLocation, std::string const&) + 0x2a (0x7f2223c0ed7a in /opt/conda/envs/lab42/lib/python3.7/site-packages/torch/lib/libc10.so)
frame #2: <unknown function> + 0xbec7e5 (0x7f2222f067e5 in /opt/conda/envs/lab42/lib/python3.7/site-packages/torch/lib/libtorch.so.1)
frame #3: void std::vector<torch::jit::fuser::PartitionDesc, std::allocator<torch::jit::fuser::PartitionDesc> >::_M_emplace_back_aux<torch::jit::fuser::TensorDesc const&, unsigned long, long const&>(torch::jit::fuser::TensorDesc const&, unsigned long&&, long const&) + 0x90 (0x7f2222f07480 in /opt/conda/envs/lab42/lib/python3.7/site-packages/torch/lib/libtorch.so.1)
frame #4: torch::jit::fuser::compileKernel(torch::jit::fuser::KernelSpec const&, torch::jit::fuser::ArgSpec const&, std::vector<long, std::allocator<long> > const&, c10::Device) + 0x125f (0x7f2222f045cf in /opt/conda/envs/lab42/lib/python3.7/site-packages/torch/lib/libtorch.so.1)
frame #5: torch::jit::fuser::runFusion(long, std::vector<c10::IValue, std::allocator<c10::IValue> >&, std::string*) + 0x5b0 (0x7f2222f0bc20 in /opt/conda/envs/lab42/lib/python3.7/site-packages/torch/lib/libtorch.so.1)
frame #6: torch::jit::runFusion(long, std::vector<c10::IValue, std::allocator<c10::IValue> >&) + 0x13 (0x7f2222e3bbc3 in /opt/conda/envs/lab42/lib/python3.7/site-packages/torch/lib/libtorch.so.1)
frame #7: <unknown function> + 0xb2b066 (0x7f2222e45066 in /opt/conda/envs/lab42/lib/python3.7/site-packages/torch/lib/libtorch.so.1)
frame #8: <unknown function> + 0xa8ebe6 (0x7f2222da8be6 in /opt/conda/envs/lab42/lib/python3.7/site-packages/torch/lib/libtorch.so.1)
frame #9: torch::jit::InterpreterState::run(std::vector<c10::IValue, std::allocator<c10::IValue> >&) + 0x22 (0x7f2222da4202 in /opt/conda/envs/lab42/lib/python3.7/site-packages/torch/lib/libtorch.so.1)
frame #10: <unknown function> + 0xa7685d (0x7f2222d9085d in /opt/conda/envs/lab42/lib/python3.7/site-packages/torch/lib/libtorch.so.1)
frame #11: <unknown function> + 0x4584f7 (0x7f226c4d14f7 in /opt/conda/envs/lab42/lib/python3.7/site-packages/torch/lib/libtorch_python.so)
frame #12: <unknown function> + 0x130fac (0x7f226c1a9fac in /opt/conda/envs/lab42/lib/python3.7/site-packages/torch/lib/libtorch_python.so)
frame #13: _PyMethodDef_RawFastCallDict + 0x265 (0x5652ed4e3a55 in /opt/conda/envs/lab42/bin/python)
frame #14: _PyCFunction_FastCallDict + 0x21 (0x5652ed4e3bd1 in /opt/conda/envs/lab42/bin/python)
frame #15: _PyObject_Call_Prepend + 0xde (0x5652ed4e19be in /opt/conda/envs/lab42/bin/python)
frame #16: PyObject_Call + 0x6e (0x5652ed4d4b9e in /opt/conda/envs/lab42/bin/python)
frame #17: <unknown function> + 0x17515a (0x5652ed52015a in /opt/conda/envs/lab42/bin/python)
frame #18: PyObject_Call + 0x6e (0x5652ed4d4b9e in /opt/conda/envs/lab42/bin/python)
frame #19: _PyEval_EvalFrameDefault + 0x1e35 (0x5652ed56d9f5 in /opt/conda/envs/lab42/bin/python)
frame #20: _PyEval_EvalCodeWithName + 0x2f9 (0x5652ed4c56f9 in /opt/conda/envs/lab42/bin/python)
frame #21: _PyFunction_FastCallDict + 0x1d5 (0x5652ed4c6805 in /opt/conda/envs/lab42/bin/python)
frame #22: _PyObject_Call_Prepend + 0xde (0x5652ed4e19be in /opt/conda/envs/lab42/bin/python)
frame #23: <unknown function> + 0x17512a (0x5652ed52012a in /opt/conda/envs/lab42/bin/python)
frame #24: _PyObject_FastCallKeywords + 0x48b (0x5652ed52118b in /opt/conda/envs/lab42/bin/python)
frame #25: _PyEval_EvalFrameDefault + 0x4a66 (0x5652ed570626 in /opt/conda/envs/lab42/bin/python)
frame #26: _PyEval_EvalCodeWithName + 0xc30 (0x5652ed4c6030 in /opt/conda/envs/lab42/bin/python)
frame #27: _PyFunction_FastCallKeywords + 0x387 (0x5652ed50b917 in /opt/conda/envs/lab42/bin/python)
frame #28: _PyEval_EvalFrameDefault + 0x14e6 (0x5652ed56d0a6 in /opt/conda/envs/lab42/bin/python)
frame #29: _PyEval_EvalCodeWithName + 0xc30 (0x5652ed4c6030 in /opt/conda/envs/lab42/bin/python)
frame #30: _PyFunction_FastCallKeywords + 0x387 (0x5652ed50b917 in /opt/conda/envs/lab42/bin/python)
frame #31: _PyEval_EvalFrameDefault + 0x14e6 (0x5652ed56d0a6 in /opt/conda/envs/lab42/bin/python)
frame #32: _PyEval_EvalCodeWithName + 0x2f9 (0x5652ed4c56f9 in /opt/conda/envs/lab42/bin/python)
frame #33: _PyFunction_FastCallKeywords + 0x387 (0x5652ed50b917 in /opt/conda/envs/lab42/bin/python)
frame #34: _PyEval_EvalFrameDefault + 0x14e6 (0x5652ed56d0a6 in /opt/conda/envs/lab42/bin/python)
frame #35: _PyEval_EvalCodeWithName + 0x2f9 (0x5652ed4c56f9 in /opt/conda/envs/lab42/bin/python)
frame #36: _PyFunction_FastCallDict + 0x400 (0x5652ed4c6a30 in /opt/conda/envs/lab42/bin/python)
frame #37: _PyEval_EvalFrameDefault + 0x1e35 (0x5652ed56d9f5 in /opt/conda/envs/lab42/bin/python)
frame #38: _PyEval_EvalCodeWithName + 0xc30 (0x5652ed4c6030 in /opt/conda/envs/lab42/bin/python)
frame #39: _PyFunction_FastCallKeywords + 0x387 (0x5652ed50b917 in /opt/conda/envs/lab42/bin/python)
frame #40: _PyEval_EvalFrameDefault + 0x14e6 (0x5652ed56d0a6 in /opt/conda/envs/lab42/bin/python)
frame #41: _PyEval_EvalCodeWithName + 0x2f9 (0x5652ed4c56f9 in /opt/conda/envs/lab42/bin/python)
frame #42: PyEval_EvalCodeEx + 0x44 (0x5652ed4c65f4 in /opt/conda/envs/lab42/bin/python)
frame #43: PyEval_EvalCode + 0x1c (0x5652ed4c661c in /opt/conda/envs/lab42/bin/python)
frame #44: <unknown function> + 0x1d08bd (0x5652ed57b8bd in /opt/conda/envs/lab42/bin/python)
frame #45: _PyMethodDef_RawFastCallKeywords + 0xe9 (0x5652ed50bf99 in /opt/conda/envs/lab42/bin/python)
frame #46: _PyCFunction_FastCallKeywords + 0x21 (0x5652ed50c231 in /opt/conda/envs/lab42/bin/python)
frame #47: _PyEval_EvalFrameDefault + 0x4764 (0x5652ed570324 in /opt/conda/envs/lab42/bin/python)
frame #48: _PyGen_Send + 0x2a2 (0x5652ed521e42 in /opt/conda/envs/lab42/bin/python)
frame #49: _PyEval_EvalFrameDefault + 0x1a87 (0x5652ed56d647 in /opt/conda/envs/lab42/bin/python)
frame #50: _PyGen_Send + 0x2a2 (0x5652ed521e42 in /opt/conda/envs/lab42/bin/python)
frame #51: _PyEval_EvalFrameDefault + 0x1a87 (0x5652ed56d647 in /opt/conda/envs/lab42/bin/python)
frame #52: _PyGen_Send + 0x2a2 (0x5652ed521e42 in /opt/conda/envs/lab42/bin/python)
frame #53: _PyMethodDef_RawFastCallKeywords + 0x8d (0x5652ed50bf3d in /opt/conda/envs/lab42/bin/python)
frame #54: _PyMethodDescr_FastCallKeywords + 0x4f (0x5652ed520c6f in /opt/conda/envs/lab42/bin/python)
frame #55: _PyEval_EvalFrameDefault + 0x4c5c (0x5652ed57081c in /opt/conda/envs/lab42/bin/python)
frame #56: _PyFunction_FastCallKeywords + 0xfb (0x5652ed50b68b in /opt/conda/envs/lab42/bin/python)
frame #57: _PyEval_EvalFrameDefault + 0x416 (0x5652ed56bfd6 in /opt/conda/envs/lab42/bin/python)
frame #58: _PyFunction_FastCallKeywords + 0xfb (0x5652ed50b68b in /opt/conda/envs/lab42/bin/python)
frame #59: _PyEval_EvalFrameDefault + 0x6a0 (0x5652ed56c260 in /opt/conda/envs/lab42/bin/python)
frame #60: _PyEval_EvalCodeWithName + 0x2f9 (0x5652ed4c56f9 in /opt/conda/envs/lab42/bin/python)
frame #61: _PyFunction_FastCallDict + 0x400 (0x5652ed4c6a30 in /opt/conda/envs/lab42/bin/python)
frame #62: _PyObject_Call_Prepend + 0x63 (0x5652ed4e1943 in /opt/conda/envs/lab42/bin/python)
frame #63: PyObject_Call + 0x6e (0x5652ed4d4b9e in /opt/conda/envs/lab42/bin/python)
```

## Environment

MacOS, Pytorch 1.1.0

## Additional context

cc @suo

**Comment by ailzhang** [comment]
@msieb1 Can you provide a minimal repro? Thanks!

**Comment by msieb1** [comment]
[solved] The problem was somewhere in the code `torch.cat` was applied to a single tensor only instead of a list of tensors, i.e., `torch.cat([tensor])`. While this worked fine for normal runs, the JIT tracer threw the above mentioned error without any hint on where it occured. @ailzhang

**Comment by zeionara** [comment]
Having the same issue when running [this model](https://huggingface.co/intfloat/multilingual-e5-large) with `sentence-transformers`:

```
SentenceTransformer(
  (0): Transformer({'max_seq_length': 512, 'do_lower_case': False}) with Transformer model: XLMRobertaModel
  (1): Pooling({'word_embedding_dimension': 1024, 'pooling_mode_cls_token': False, 'pooling_mode_mean_tokens': True, 'pooling_mode_max_tokens': False, 'pooling_mode_mean_sqrt_len_tokens': False, 'pooling_mode_weightedmean_tokens': False, 'pooling_mode_lasttoken': False, 'include_prompt': True})
  (2): Normalize()
)
```

On `triton` inference server:

```
PyTorch execute failure: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript (most recent call last):
RuntimeError: nSubTensors_ > 1 INTERNAL ASSERT FAILED at "/opt/pytorch/pytorch/torch/csrc/jit/codegen/fuser/partition_desc.h":24, please report a bug to PyTorch.
```

The `triton` container version is `24.02`, without `triton` the model works fine

**Comment by zeionara** [comment]
> Having the same issue when running [this model](https://huggingface.co/intfloat/multilingual-e5-large) with `sentence-transformers`:
> 
> ```
> SentenceTransformer(
>   (0): Transformer({'max_seq_length': 512, 'do_lower_case': False}) with Transformer model: XLMRobertaModel
>   (1): Pooling({'word_embedding_dimension': 1024, 'pooling_mode_cls_token': False, 'pooling_mode_mean_tokens': True, 'pooling_mode_max_tokens': False, 'pooling_mode_mean_sqrt_len_tokens': False, 'pooling_mode_weightedmean_tokens': False, 'pooling_mode_lasttoken': False, 'include_prompt': True})
>   (2): Normalize()
> )
> ```
> 
> On `triton` inference server:
> 
> ```
> PyTorch execute failure: The following operation failed in the TorchScript interpreter.
> Traceback of TorchScript (most recent call last):
> RuntimeError: nSubTensors_ > 1 INTERNAL ASSERT FAILED at "/opt/pytorch/pytorch/torch/csrc/jit/codegen/fuser/partition_desc.h":24, please report a bug to PyTorch.
> ```
> 
> The `triton` container version is `24.02`, without `triton` the model works fine

The problem seems to be solved by switching from `triton` version `24.02` to `22.12`
